### PR TITLE
Added apparmor profile for redshift

### PR DESCRIPTION
**Pull request by CameronNemo**

### CODE REVIEW

**Comment by jonls** [comment]
Thanks! Since I am not particularly familiar with AppArmor, please take a moment to explain what this pull request does and why you think it should be included in Redshift.

**Comment by CameronNemo** [comment]
AppArmor is a Mandatory Access Control mechanism: instead of considering users and groups, like traditional Unix permissions, it considers what process is making the request. More basically, it is a simple way to increase confinement of Redshift by restricting what files redshift can access and other permissions.

With the AppArmor profile in enforcement mode, redshift will not be able to access the users normal files in $HOME -- only its own config file. Furthermore it will be restricted from many other areas of the system that are not necessary to its operation.

File: configure.ac [hunk 1]
```diff
@@ -335,6 +335,19 @@ AS_IF([test -n "$with_systemduserunitdir" -a "x$with_systemduserunitdir" != xno]
 AM_CONDITIONAL([ENABLE_SYSTEMD], [test "x$enable_systemd" != xno])
 
 
+# Check for AppArmor
+AC_MSG_CHECKING([whether to enable AppArmor profile])
+AC_ARG_ENABLE([apparmor], [AC_HELP_STRING([--enable-apparmor],
+	[enable AppArmor profile])],
+        [enable_apparmor=$enableval],[enable_apparmor=no])
+AS_IF([test "x$enable_apparmor" != xno], [
+        AC_MSG_RESULT([yes])
```
Comment (jonls) [inline configure.ac]: `enable_apparmor=yes`

File: configure.ac [hunk 2]
```diff
@@ -335,6 +335,19 @@ AS_IF([test -n "$with_systemduserunitdir" -a "x$with_systemduserunitdir" != xno]
...
+], [
+        AC_MSG_RESULT([no])
```
Comment (jonls) [inline configure.ac]: `enable_apparmor=no`

**Comment by jonls** [comment]
Ok, thanks! I added some minor suggestions for `configure.ac`.

**Comment by jonls** [comment]
It's been more than a year with no progress. Since there hasn't been interest in getting this merged I will close this. If someone wants to take this up again, please reopen.

**Comment by CameronNemo** [comment]
@jonls I am unclear on what your comments are suggesting. Do you want me to change the message to `enable_apparmor=...`?

**Comment by jonls** [comment]
@CameronNemo The variable `enable_apparmor` should be explicitly set to `yes` or `no`. See for example the section right above that sets `enable_systemd` to either `yes` or `no`: https://github.com/jonls/redshift/pull/227/files#diff-67e997bcfdac55191033d57a16d1408aL320

**Pull request closed without merging**